app/main.py

```diff
@@ -1 +1,36 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False
+                 ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        if self.health > 0:
+            Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name},"
+                f" Health: {self.health},"
+                f" Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    @staticmethod
+    def bite(herbivore_animal: Herbivore) -> None:
+        check_class = isinstance(herbivore_animal, Herbivore)
+        check_hidden_value = herbivore_animal.hidden
+        if check_class and not check_hidden_value:
```

Comment:
you can create additional variables for readability but I suggest better naming
`if is_herbivore_instance and not is_hidden:`
they tell us more info